### DISCUSSION TITLE
How to add CRUD options to permission

### DISCUSSION THREAD

**Discussion by boutamente-abdessamad** [spatie/laravel-permission#2008] [Q&A] [answered]
i work in project with laravel and i used `spatie/laravel-permission` to manage permissions in my app the package work perfectly with the standard way, but i need something different ,i need to add CRUD options for each permission
what is the best way  to add permissions options like image below without Loss package helpers
![image](https://user-images.githubusercontent.com/35835561/152044641-e70c69f3-6142-4356-b84c-cc6625bc7aa9.png)

**Comment by erikn69** [comment, marked as answer]
There is a examples for CRUD of permissions: **[ui-options](https://github.com/spatie/laravel-permission/blob/main/docs/advanced-usage/ui-options.md)**
>## Need a UI?
>
>The package doesn't come with any screens out of the box, you should build that yourself. Here are some options to get you started:
>
>- If you'd like to build your own UI, and understand the underlying logic for Gates and Roles and Users, the [Laravel 6  User Login and Management With Roles](https://www.youtube.com/watch?v=7PpJsho5aak&list=PLxFwlLOncxFLazmEPiB4N0iYc3Dwst6m4) video series by Mark Twigg of Penguin Digital gives thorough coverage to the topic, the theory, and implementation of a basic Roles system, independent of this Permissions Package.
>
>- [Laravel Nova package by @vyuldashev for managing Roles and Permissions](https://github.com/vyuldashev/nova-permission)
>
>- [Laravel Nova package by @paras-malhotra for managing Roles and Permissions and permissions based authorization for Nova Resources](https://github.com/insenseanalytics/laravel-nova-permission)
>
>- [How to create a UI for managing the permissions and roles](http://www.qcode.in/easy-roles-and-permissions-in-laravel-5-4/)
>
>- [Laravel User Management for managing users, roles, permissions, departments and authorization](https://github.com/Mekaeil/LaravelUserManagement) by [Mekaeil](https://github.com/Mekaeil)
>
>- [Generating UI boilerplate using InfyOm](https://youtu.be/hlGu2pa1bdU) video tutorial by [Shailesh](https://github.com/shailesh-ladumor)
>
>- [LiveWire Base Admin Panel](https://github.com/alighasemzadeh/bap) User management by [AliGhasemzadeh](https://github.com/alighasemzadeh)

**Reply by boutamente-abdessamad** [reply]
thank you for your help
